pkg/arch/arch.go

```diff
@@ -5,91 +5,104 @@ import (
 	"runtime"
 )
 
-var ErrUnsupportedArch = fmt.Errorf("unsupported architecture: %s", runtime.GOARCH)
+type Arch struct {
+	arch string
+}
+
+func NewArch(arch string) (*Arch, error) {
+	switch arch {
+	case "amd64", "arm64":
+		return &Arch{arch}, nil
+	default:
+		return nil, fmt.Errorf("unsupported architecture %s", arch)
+	}
+}
 
 // Target returns the Linux Makefile target to build the kernel, for historical
 // reasons, those are different between architectures.
-func Target(arch string) (string, error) {
-	if arch == "" {
-		arch = runtime.GOARCH
-	}
-	switch arch {
+func (a *Arch) Target() string {
```

Comment:
nit: I would use a value receiver in all those methods since we don't modify arch.

Reply:
Makes sense, it's just a string anyway. Fixed.